### ISSUE TITLE
Possible bug: broken OpenDroneMap snap on core20

### ISSUE THREAD

**Issue by Saijin-Naib** [snapcore/core20#128]
Local reproduction:  
```
 sudo snap install opendronemap --edge
```
```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal
```
```
uname -r
5.10.60.1-microsoft-standard-WSL2
```
```
snap version
snap    2.53.4
snapd   2.53.4
series  16
ubuntu  20.04
kernel  5.10.60.1-microsoft-standard-WSL2
```
```
snap run opendronemap
Python path configuration:
  PYTHONHOME = (not set)
  PYTHONPATH = (not set)
  program name = 'python3'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = '/usr/bin/python3'
  sys.base_prefix = '/usr'
  sys.base_exec_prefix = '/usr'
  sys.executable = '/usr/bin/python3'
  sys.prefix = '/usr'
  sys.exec_prefix = '/usr'
  sys.path = [
    '/usr/lib/python38.zip',
    '/usr/lib/python3.8',
    '/usr/lib/python3.8/lib-dynload',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00007f2e195c7740 (most recent call first):
<no Python frame>
```

In speaking with Dani, they indicated that they have seen similar breakage even with their own snaps and they recommended that we downgrade our core to core18 and then adjust the stage-packages to reflect this.

Our snapcraft yaml(s):  
https://github.com/OpenDroneMap/ODM/blob/master/snap/snapcraft.yaml

User threads (other reports/reproductions):  
https://community.opendronemap.org/t/snap-opendronemap-python-fail-on-ubuntu-21/9949?u=saijin_naib
https://community.opendronemap.org/t/odm-and-ubuntu-20-04/8223?u=saijin_naib
https://community.opendronemap.org/t/issue-with-opensfm-reconstruction/7381/11?u=saijin_naib

Does anyone have any further guidance on this, or ways that I can collect more information if this is a confirmed bug?

**Comment by anonymouse64** [comment]
Can you reproduce the bug on a supported version of Ubuntu like one in a VM not in WSL2? WSL2 is not supported by snapd at this time

**Comment by Saijin-Naib** [comment]
> Can you reproduce the bug on a supported version of Ubuntu like one in a VM not in WSL2? WSL2 is not supported by snapd at this time

Do the user reports/reproductions not count?

**Comment by anonymouse64** [comment]
Sorry I just looked at your report and saw:
```
kernel  5.10.60.1-microsoft-standard-WSL2
```

which is not a supported target for snapd. 

I did just try installing the snap on my 21.10 system and can reproduce the issue so it doesn't seem WSL2 specific. I don't see any obvious confinement related things like AppArmor or Seccomp denials so it's probably not anything related to the snap sandbox. 

More folks may help you debug this issue if you report it on the snapcraft forum at forum.snapcraft.io and link back to this bug report
